solc/CommandLineInterface.cpp

```diff
@@ -594,6 +594,29 @@ bool CommandLineInterface::compile()
 
 		m_compiler->enableIRGeneration(m_options.compiler.outputs.ir || m_options.compiler.outputs.irOptimized);
 		m_compiler->enableEwasmGeneration(m_options.compiler.outputs.ewasm);
+		m_compiler->enableEvmBytecodeGeneration(
+			m_options.compiler.outputs.asm_ ||
+			m_options.compiler.outputs.asmJson ||
+			m_options.compiler.outputs.opcodes ||
+			m_options.compiler.outputs.binary ||
+			m_options.compiler.outputs.binaryRuntime ||
+			m_options.compiler.outputs.metadata ||
+			(
```

Comment:
```suggestion
			m_options.compiler.outputs.metadata || (
```

Reply:
Not sure I correctly understand the comment, but it looks inconsistent with the coding style to me, doesn't it?
The block in parenthesis is basically the same as any of the previous clauses, why should I *not* break before it here, even though I have broken all other ``||`` clauses?
Funnily I would have skipped these parentheses entirely, but the compiler won't let me, even though it'd be correct :-).

Reply:
>  why should I not break before it here, even though I have broken all other `||` clauses?

Well, that's the style that feels intuitively correct to me and you even used it yourself on the `m_options.compiler.combinedJsonRequests && (` line, but if you want a more concrete justification, here's what the [CODING_STYLE](https://github.com/ethereum/solidity/blob/develop/CODING_STYLE.md) says:

> > If lines are broken, a list of elements enclosed with parentheses (of any kind) and separated by a separator (of any kind) are formatted such that there is exactly one element per line, followed by the separator, the opening parenthesis is on the first line, followed by a line break and the closing parenthesis is on a line of its own unindented). See example below.

This matches how we wrap function calls, `if`s and other parentheses as far as I can tell.

> Funnily I would have skipped these parentheses entirely, but the compiler won't let me, even though it'd be correct :-).

Yeah, I hate that too. I can see how operator precedence can be confusing in more obscure cases, and I sometimes add extra parentheses myself, but I don't understand why nowadays many compiler and linters insist on requiring parentheses in very clear cases like `&&` vs `||`. I've seen that tendency in Python too and it's really annoying :)

Reply:
I mean, I really don't mind moving the parentheses here :-).
But I do in fact read the coding style differently here and the other line you're quoting is a different case in my eyes.

I.e. I start from:

```
m_compiler->enableEvmBytecodeGeneration(m_options.compiler.estimateGas || m_options.compiler.outputs.asm_ || m_options.compiler.outputs.asmJson || m_options.compiler.outputs.opcodes || m_options.compiler.outputs.binary || 	m_options.compiler.outputs.binaryRuntime || (m_options.compiler.combinedJsonRequests && (m_options.compiler.combinedJsonRequests->binary || m_options.compiler.combinedJsonRequests->binaryRuntime || m_options.compiler.combinedJsonRequests->opcodes || m_options.compiler.combinedJsonRequests->asm_ || m_options.compiler.combinedJsonRequests->generatedSources || m_options.compiler.combinedJsonRequests->generatedSourcesRuntime || m_options.compiler.combinedJsonRequests->srcMap || m_options.compiler.combinedJsonRequests->srcMapRuntime || m_options.compiler.combinedJsonRequests->funDebug || m_options.compiler.combinedJsonRequests->funDebugRuntime)));
```
Now I break the outermost parentheses. That means, the initial ``(`` is on one line, then one element each line followed by the separator and a line break. So I get:
```
m_compiler->enableEvmBytecodeGeneration(
	m_options.compiler.estimateGas ||
	m_options.compiler.outputs.asm_ ||
	m_options.compiler.outputs.asmJson ||
	m_options.compiler.outputs.opcodes ||
	m_options.compiler.outputs.binary ||
	m_options.compiler.outputs.binaryRuntime ||
	(m_options.compiler.combinedJsonRequests && (m_options.compiler.combinedJsonRequests->binary || m_options.compiler.combinedJsonRequests->binaryRuntime || m_options.compiler.combinedJsonRequests->opcodes || m_options.compiler.combinedJsonRequests->asm_ || m_options.compiler.combinedJsonRequests->generatedSources || m_options.compiler.combinedJsonRequests->generatedSourcesRuntime || m_options.compiler.combinedJsonRequests->srcMap || m_options.compiler.combinedJsonRequests->srcMapRuntime || m_options.compiler.combinedJsonRequests->funDebug || m_options.compiler.combinedJsonRequests->funDebugRuntime))
);
```

The inner parentheses are one element in the ``||``-separated list, so they do get their own line.

Now I break the ``&&`` separated list in the inner parentheses. Same thing again: the ``(`` stays on the first line, then one element each line followed by the separator and line break:

```
m_compiler->enableEvmBytecodeGeneration(
	m_options.compiler.estimateGas ||
	m_options.compiler.outputs.asm_ ||
	m_options.compiler.outputs.asmJson ||
	m_options.compiler.outputs.opcodes ||
	m_options.compiler.outputs.binary ||
	m_options.compiler.outputs.binaryRuntime ||
	(
		m_options.compiler.combinedJsonRequests &&
		(m_options.compiler.combinedJsonRequests->binary || m_options.compiler.combinedJsonRequests->binaryRuntime || m_options.compiler.combinedJsonRequests->opcodes || m_options.compiler.combinedJsonRequests->asm_ || m_options.compiler.combinedJsonRequests->generatedSources || m_options.compiler.combinedJsonRequests->generatedSourcesRuntime || m_options.compiler.combinedJsonRequests->srcMap || m_options.compiler.combinedJsonRequests->srcMapRuntime || m_options.compiler.combinedJsonRequests->funDebug || m_options.compiler.combinedJsonRequests->funDebugRuntime)
	)
);
```




Now I further break the innermost ``||``-separated list and get this:


```
m_compiler->enableEvmBytecodeGeneration(
	m_options.compiler.estimateGas ||
	m_options.compiler.outputs.asm_ ||
	m_options.compiler.outputs.asmJson ||
	m_options.compiler.outputs.opcodes ||
	m_options.compiler.outputs.binary ||
	m_options.compiler.outputs.binaryRuntime ||
	(
		m_options.compiler.combinedJsonRequests &&
		(
			m_options.compiler.combinedJsonRequests->binary ||
			m_options.compiler.combinedJsonRequests->binaryRuntime ||
			m_options.compiler.combinedJsonRequests->opcodes ||
			m_options.compiler.combinedJsonRequests->asm_ ||
			m_options.compiler.combinedJsonRequests->generatedSources ||
			m_options.compiler.combinedJsonRequests->generatedSourcesRuntime ||
			m_options.compiler.combinedJsonRequests->srcMap ||
			m_options.compiler.combinedJsonRequests->srcMapRuntime ||
			m_options.compiler.combinedJsonRequests->funDebug ||
			m_options.compiler.combinedJsonRequests->funDebugRuntime
		)
	)
);
```

Granted that's not exactly what I have in the PR, but it's not ``m_options.compiler.outputs.binaryRuntime || (\n`` either.

I could also *not* have broken the ``&&``-separated list, then I'd get:

```
m_compiler->enableEvmBytecodeGeneration(
	m_options.compiler.estimateGas ||
	m_options.compiler.outputs.asm_ ||
	m_options.compiler.outputs.asmJson ||
	m_options.compiler.outputs.opcodes ||
	m_options.compiler.outputs.binary ||
	m_options.compiler.outputs.binaryRuntime ||
	(m_options.compiler.combinedJsonRequests && (
		m_options.compiler.combinedJsonRequests->binary ||
		m_options.compiler.combinedJsonRequests->binaryRuntime ||
		m_options.compiler.combinedJsonRequests->opcodes ||
		m_options.compiler.combinedJsonRequests->asm_ ||
		m_options.compiler.combinedJsonRequests->generatedSources ||
		m_options.compiler.combinedJsonRequests->generatedSourcesRuntime ||
		m_options.compiler.combinedJsonRequests->srcMap ||
		m_options.compiler.combinedJsonRequests->srcMapRuntime ||
		m_options.compiler.combinedJsonRequests->funDebug ||
		m_options.compiler.combinedJsonRequests->funDebugRuntime
	))
);
```

I'll probably change the PR to that.

But I don't think your suggestion would actually follow the coding style - though I also wouldn't say that the coding style is overly clear on this :-).

Reply:
I wonder how many inconsistencies in the code in these matters would be noticed by some kind of linting tool if we had any that supported this stuff properly :-).